e2e/testenv/githelper/git.go

```diff
@@ -190,46 +193,40 @@ func (g *Git) Create(repodir string, from string, subdir string) (*git.Repositor
 		return nil, err
 	}
 
-	w, err := repo.Worktree()
-	if err != nil {
-		return nil, err
-	}
+	_, err = g.Update(repo, UpdateForce)
 
-	if _, err := w.Add("."); err != nil {
-		return nil, err
-	}
+	return repo, err
+}
 
-	_, err = w.Commit("add chart", &git.CommitOptions{
-		Author: author(),
-	})
+// Add adds files to a repo, commits them and pushed the changes.
+func (g *Git) Add(repodir, from, subdir string) (string, error) {
+	s := osfs.New(path.Join(repodir, ".git"))
+	storer := filesystem.NewStorage(s, cache.NewObjectLRUDefault())
+	fs := osfs.New(repodir)
+	repo, err := git.Open(storer, fs)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
 
-	po := git.PushOptions{
-		Progress: os.Stdout,
-		// Force push, so our initial state is deterministic
-		RefSpecs: []config.RefSpec{config.RefSpec("+refs/heads/master:refs/heads/" + g.Branch)},
-		// This prevents IP SANs from being needed on the git server cert; the TLS verification we are most
-		// interested in is the one happening in Fleet's controllers and jobs.
-		InsecureSkipTLS: true,
-	}
-	k, err := g.Auth.getKeys()
+	cmd := exec.Command("cp", "-a", from, path.Join(repodir, subdir)) //nolint:gosec // test code should never receive user input
```

Comment:
Do we need to shell out here?
Would something like [os.CopyFS](https://pkg.go.dev/os@go1.25.3#CopyFS) work instead?